File: src/main/java/org/fletchly/genius/ollama/model/OllamaRequest.java
package org.fletchly.genius.ollama.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Builder;
import lombok.Singular;
import org.jetbrains.annotations.NotNull;

import java.util.List;

@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public record OllamaRequest(
        @NotNull
        String model,
        @NotNull
        @Singular
        List<OllamaMessage> messages,
        @NotNull
        OllamaOptions options,
        boolean stream,
        boolean think
) {
}

File: src/main/kotlin/io/fletchly/genius/ollama/model/OllamaRequest.kt
package io.fletchly.genius.ollama.model

import io.fletchly.genius.conversation.model.Message
import kotlinx.serialization.Serializable

@Serializable
data class OllamaRequest(
    val model: String,
    val messages: List<Message>,
    val options: OllamaOptions,
    val stream: Boolean = false,
    val think: Boolean = false
)